### PR TITLE
[SYCLomatic][Driver] Enabled migration of cuPointerGetAttributes

### PR DESCRIPTION
**Pull request by TejaX-Alaghari**
Added migration support for cuPointerGetAttributes API

### CODE REVIEW

File: clang/runtime/dpct-rt/include/dpct/memory.hpp [hunk 1]
```diff
@@ -1603,6 +1603,23 @@ template <typename T>
 using usm_device_allocator = detail::deprecated::usm_allocator<T, sycl::usm::alloc::shared>;
 } // namespace deprecated
 
+enum class attribute_type {
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/dpct/memory.hpp]: Can we this enum class into `pointer_attributes`?

File: clang/runtime/dpct-rt/include/dpct/memory.hpp [hunk 2]
```diff
@@ -1603,6 +1603,23 @@ template <typename T>
...
+  p2p_tokens,
+  sync_memps,
+  buffer_id,
+  is_managed,
+  device_id,
+  is_legacy_ipc_capable,
+  range_start_addr,
+  range_size,
+  mapped,
+  allowed_handle_types
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/dpct/memory.hpp]: I think adding the 1:1 attributes, which we have mapping, is enough.
And add one more "unsupported" enum value to map all the unsupported attributes.

File: clang/runtime/dpct-rt/include/dpct/memory.hpp (hunk 1 above)
Comment (TejaX-Alaghari) [reply clang/runtime/dpct-rt/include/dpct/memory.hpp]: Moved to pointer_attributes::type

**Review by zhiweij1 — APPROVED** [review]

**Review by AndyCHHuang — APPROVED** [review]
LGTM

**Review by ziranzha — APPROVED** [review]

File: clang/lib/DPCT/APINames.inc [hunk 3]
```diff
@@ -1763,7 +1763,7 @@ ENTRY(cuMemPrefetchAsync, cuMemPrefetchAsync, true, NO_FLAG, P4, "Partial: USM o
 ENTRY(cuMemPrefetchAsync_v2, cuMemPrefetchAsync_v2, false, NO_FLAG, P4, "Partial: USM only, DPCT1007")
 ENTRY(cuMemRangeGetAttribute, cuMemRangeGetAttribute, false, NO_FLAG, P4, "comment")
 ENTRY(cuMemRangeGetAttributes, cuMemRangeGetAttributes, false, NO_FLAG, P4, "comment")
-ENTRY(cuPointerGetAttribute, cuPointerGetAttribute, false, NO_FLAG, P4, "comment")
+ENTRY(cuPointerGetAttribute, cuPointerGetAttribute, true, NO_FLAG, P4, "Successful")
```
Comment (zhimingwang36) [inline clang/lib/DPCT/APINames.inc]: Successful  ==>  Partial?

File: clang/runtime/dpct-rt/include/dpct/memory.hpp [hunk 4]
```diff
@@ -1676,6 +1685,40 @@ class pointer_attributes {
 #endif
   }
 
+  static void get(unsigned int numAttributes, type *attributes,
```
Comment (zhimingwang36) [inline clang/runtime/dpct-rt/include/dpct/memory.hpp:1697]: Please add comment for this publich interface, this function is not so straightforward.

**Review by zhimingwang36 — APPROVED** [review]

**Pull request merged**